src/D2L.Security.OAuth2/Provisioning/AccessTokenProviderFactory.cs

```diff
@@ -18,18 +19,34 @@ public static IAccessTokenProvider Create(
 			ITokenSigner tokenSigner,
 			HttpClient httpClient,
 			Uri authEndpoint,
-			TimeSpan tokenRefreshGracePeriod
+			TimeSpan tokenRefreshGracePeriod,
+			IAccessTokenProvider inner = null,
+			ICache cache = null
 		) {
+			if( inner != null && cache == null ) {
+				throw new InvalidOperationException( "If you provide an inner you need to also provide its cache" );

```

Comment:
The idea here is that this factory is either for wrapping your IAccessTokenProvider in our caching layer (for which you provide an ICache for us to use) or for using our jwt-bearer-grant based impl with or without caching.

The reason I'm doing this is I want to implement an IamGrant IAccessTokenProvider impl. I want to do that inside the LMS because I'd really like to use the LMS Lambda framework. It would make sense to implement it in this library, and I mean you could at least define the input/output here, but meh.